### ISSUE TITLE
Some tests fail if "version" envvar is set

### ISSUE THREAD

**Issue by puzzlewolf** [ChordPro/chordpro#293]
## Describe the bug
Hi :) I'm working on updating the chordpro package in NixOS to 6.010. Some tests (`320_subst.t`, `321_subst.t`, `322_subst.t`) fail during the package's test phase, I think because NixOS sets the environment variable `version` and that is read back instead of the expected configuration value for `nested.else`.

### To Reproduce
1.  In a bash shell
2. `export version=TEST_VERSION`
3. `make test`
4. Tests fail, here's the log snippet:
```
t/320_subst.t ................ 
t/320_subst.t ................ 1/2 
#   Failed test at t/320_subst.t line 19.
#          got: '# @ = version nested
# version = '1'
# # nested.@ = version something else
# nested.version = '3'
# nested.something = '3'
# nested.else = 'TEST_VERSION'
# '
#     expected: '# @ = version nested
# version = '1'
# # nested.@ = version something else
# nested.version = '3'
# nested.something = '3'
# nested.else = '1'
# '

#   Failed test at t/320_subst.t line 29.
#     Structures begin differing at:
#          $got->{nested.else} = 'TEST_VERSION'
#     $expected->{nested.else} = '1'
# Looks like you failed 2 tests of 2.

                                    
t/320_subst.t ................ Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/2 subtests 
t/321_subst.t ................ 
t/321_subst.t ................ 1/1 
#   Failed test at t/321_subst.t line 31.
#          got: '# @ = version nested XXDATAPROPERTIESXX test1 test2
# version = '1'
# # nested.@ = nothing version something else
# nested.nothing = 'xxx'
# nested.version = '3'
# nested.something = '3'
# nested.else = 'TEST_VERSION'
# XXDATAPROPERTIESXX = 'local'
# test1 = 'env'
# test2 = 'local'
# '
#     expected: '# @ = version nested XXDATAPROPERTIESXX test1 test2
# version = '1'
# # nested.@ = nothing version something else
# nested.nothing = 'xxx'
# nested.version = '3'
# nested.something = '3'
# nested.else = '1'
# XXDATAPROPERTIESXX = 'local'
# test1 = 'env'
# test2 = 'local'
# '
# Looks like you failed 1 test of 1.

                                    
t/321_subst.t ................ Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
t/322_subst.t ................ 
t/322_subst.t ................ 1/3 
#   Failed test 'v1'
#   at t/322_subst.t line 11.
#          got: 'TEST_VERSION'
#     expected: '1'
# Looks like you failed 1 test of 3.
                         
t/322_subst.t ................ Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/3 subtests 
```

### Expected behavior
The tests succeed :) I think if a test requires that an environment variable is unset, it should probably take care to unset it :)

## System information
 - OS: NixOS 22.11
 - Where did you get ChordPro from? CPAN
 - ChordPro version: 6.010
 - Perl version: 5.36.0

## Additional context
I guess adding `delete $ENV{version};` at the top of the tests should do the trick. I can make a PR if you're happy with that solution.

**Comment by sciurius** [comment]
Thanks for your report. What happens is clear, and there are several solutions possible (you named one of them). I'll take care of it.

I must add, however, that it is a long standing convention —from the very first inception of Unix in 1971— that environments (e.g. operating systems, desktops) should preset environment variables **in uppercase letters only**. It may be worth to file an issue with NixOS as well.

**Comment by sciurius** [comment]
Fixed in 6.010_046.

**Comment by puzzlewolf** [comment]
Thanks for the quick fix :rocket: 

The handling of environment variables is quite deep down in NixOS packaging. I'm only a satellite contributor and don't feel like the right person to bring that up. Thanks for the insight, anyway :)
